app/utilities/supplementary_data_parser.py

```diff
@@ -0,0 +1,88 @@
+from typing import Mapping
+
+from marshmallow import (
+    EXCLUDE,
+    INCLUDE,
+    Schema,
+    ValidationError,
+    fields,
+    validate,
+    validates_schema,
+)
+
+from app.authentication.auth_payload_versions import SupplementaryDataSchemaVersion
+from app.utilities.metadata_parser_v2 import VALIDATORS, StripWhitespaceMixin
+
+
+class ItemsSchema(Schema):
+    identifier = VALIDATORS["string"](validate=validate.Length(min=1))
+
+
+class ItemsData(Schema, StripWhitespaceMixin):
+    pass
+
+
+class SupplementaryData(Schema, StripWhitespaceMixin):
+    identifier = VALIDATORS["string"](validate=validate.Length(min=1))
+    schema_version = VALIDATORS["string"](
+        validate=validate.OneOf([SupplementaryDataSchemaVersion.V1.value])
+    )
+    items = fields.Nested(ItemsData, required=False, unknown=EXCLUDE)
+
+    @validates_schema()
+    def validate_unit_id(self, data, **kwargs):
+        # pylint: disable=no-self-use, unused-argument
```

Comment:
I can see this in other places, but why do we need to have kwargs then disable pylint?

Reply:
This is needed because when this validator gets called marshmallow passes some extra fields we don't use e.g. "partial" and "many"

Reply:
ahh makes sense - thanks 👍